docs/basics/part-7-using-docker-with-earthly.md

```diff
@@ -0,0 +1,541 @@
+
+### Pulling an Image
+```Dockerfile
+hello:
+    FROM earthly/dind:alpine
+    WITH DOCKER --pull hello-world
+        RUN docker run hello-world
+    END
+
+```
+You can see in the command above that we can pass a flag to `WITH DOCKER` telling it to pull an image from Docker Hub. We can pass other flags to [load in artifacts built by other targets](#loading-an-image) `--load` or even images defined by [docker-compose](#a-real-world-example) `--compose`. These images will be available within the context of `WITH DOCKER`'s docker daemon.
+
+### Loading an Image
+We can load in an image created by another target with the `--load` flag.
+
+```Dockerfile
+my-hello-world:
+    FROM ubuntu
+    CMD echo 'hello world'
+    SAVE IMAGE my-hello:latest
+
+hello:
+    FROM earthly/dind:alpine
+    WITH DOCKER --load hello:latest=+my-hello-world
+        RUN docker run hello
+    END
+```
+
+## A Real World Example
+
+One common use case for `WITH DOCKER` is running integration tests that require other services. In this case we need to set up a redis service for our tests. For this we can user a `docker-compose.yml`.
+
+`docker-compose.yml`
+```yml
+version: "3"
+services:
+  redis:
+    container_name: local-redis
+    image: redis:6.0-alpine
+    ports:
+      - 127.0.0.1:6379:6379
+    hostname: redis
+    healthcheck:
+      test: ["CMD", "curl", "-f", "http://localhost:6379"]
+      interval: 1s
+      timeout: 10s
+      retries: 5
+
+```
+`main.go`
+```go
+package main
+
+import (
+	"github.com/sirupsen/logrus"
+)
+
+var howCoolIsEarthly = "IceCool"
+
+func main() {
+	logrus.Info("hello world")
+}
+```
+`main_integration_test.go`
+```go
+package main
+
+import (
+	"context"
+	"testing"
+
+	"github.com/go-redis/redis/v8"
+	"github.com/stretchr/testify/require"
+)
+
+func TestIntegration(t *testing.T) {
+	ctx := context.Background()
+	rdb := redis.NewClient(&redis.Options{
+		Addr:     "localhost:6379",
+		Password: "", // no password set
+		DB:       0,  // use default DB
+	})
+
+	err := rdb.Set(ctx, "howCoolIsEarthly", howCoolIsEarthly, 0).Err()
+	if err != nil {
+		panic(err)
+	}
+
+	resultFromDB, err := rdb.Get(ctx, "howCoolIsEarthly").Result()
+	if err != nil {
+		panic(err)
+	}
+	require.Equal(t, howCoolIsEarthly, resultFromDB)
+}
+```
+
+```Dockerfile
+VERSION 0.6
+FROM golang:1.15-alpine3.13
+WORKDIR /go-example
+
+deps:
+    COPY go.mod go.sum ./
+    RUN go mod download
+    SAVE ARTIFACT go.mod AS LOCAL go.mod
+    SAVE ARTIFACT go.sum AS LOCAL go.sum
+
+test-setup:
+    FROM +deps
+    COPY main.go .
+    COPY main_integration_test.go .
+    ENV CGO_ENABLED=0
+    ENTRYPOINT [ "go", "test", "github.com/earthly/earthly/examples/go"]
+    SAVE IMAGE test:latest
+
+integration-tests:
+    FROM earthly/dind:alpine
+    COPY docker-compose.yml ./
+    WITH DOCKER --compose docker-compose.yml --load tests:latest=+test-setup
+        RUN docker run --network=host tests 
```

Comment:
```suggestion
        RUN docker run --network=host tests:latest
```

Reply:
We should also try to avoid `--network=host`.

Can you make this work with `--network=<name-of-compos-project>_default` - usually the name of the compose project is simply the name of the directory.